### ISSUE TITLE
cannot get version information (or help) as root

### ISSUE THREAD

**Issue by GitMensch** [microsoft/vscode#104638]
- VSCode Version: 1.48.0
- OS Version: Linux x64 3.10.0-1127.13.1.el7.x86_64

Steps to Reproduce:

1. login as root (or `sudo su`)
2. try to query the version or output the help
```sh
root@pc#/tmp> code --version
You are trying to start VSCode as a super user which isn't recommended. If this was intended, please specify an alternate user data directory using the `--user-data-dir` argument.
root@pc#/tmp> code --help
You are trying to start VSCode as a super user which isn't recommended. If this was intended, please specify an alternate user data directory using the `--user-data-dir` argument.
```

In the case of "only output information" like `--version` and `--help` the super user check (which is _very reasonable and useful in cases where vscode would be started_) should not be done.

**Comment by bpasero** [comment]
Came in via https://github.com/microsoft/vscode/commit/6456f85788a8c526d73957656aaf2a5a04091aa7 it seems.

**Comment by GitMensch** [comment]
> Came in via [6456f85](https://github.com/microsoft/vscode/commit/6456f85788a8c526d73957656aaf2a5a04091aa7) it seems.

Yes, it maybe its enough to move the block shortly before the actual start or internally hold information about "was called with output-only argument" and check this in the moved place.
